135 write to epr initials

fredkingham opened the pull request:
EPR requires a written_by field.

In TB patient consulations and other services microbiology input this is populated by the initials field. 

We change the timeline to accept standard opal template tag validation, in this case required=True. This means changing it so that after the form is submitted the form is marked as pristine. In angular terms this is not submitted and not dirty. This is so validation is reset for the next time the form is submitted.

With the covid branch its simpler. We populated the EPR initials field with the consulation from CovidFollowUpCall, CovidFollowUpCallFollowUpCall, CovidSixMonthFollowUp. We make this field required in the form, not in the model as we have existing data with None in the clinican.

Note the logic in the RNOH form is unchanged but because the API for the clinicalTimeline has change (save takes a form now), the template has been changed.

Commit:
Changes the infectious diseases timeline micro input to require initials

This changes it so that initials are required by the microbiology input
form so that when we send the data down to EPR the field is always
populated.

Commit:
Don't show the send to EPR button for RNOH clinical advice

Don't show the send to EPR button for RNOH clinical advice as they
patients are not (usually) at RNOH.

We identify RNOH clinical advice as those with a reason for clinical
interaction starting with RNOH

Commit:
Adds a comment to the infection service clinical advice timline

Adds a comment to make it clear we do show/hide rather than ng-disabled.

Its because the check-form directive also plays around with disabled
state.

Commit:
Makes initials on tb patient consultations required

Commit:
Makes the clinician in the covid 19 forms required

In order to send clinical advice down we need a written by populated.
For covid forms written by is populated by the clinician so make sure
this field is filled in by the user.

Commit:
Fixes the unit tests for the ClinicalTimeline controller

Changes it to check that the save method takes a form object and changes
it so that the form object calls setPristine

davidmiller commented on elcid/templates/inline_forms/clinical_timeline.html:
```diff
@@ -72,6 +72,6 @@
 
 {% block send_upstream_button %}
-<i class="fa fa-exchange edit pull-right pointer"
+<i ng-show="item.reason_for_interaction && item.reason_for_interaction.indexOf('RNOH') !== 0" class="fa fa-exchange edit pull-right pointer"
```
Wouldn't a `-1` spelling of this be safer against future spellings of reason for interaction?

fredkingham replied:
I thought I was future proofing by making sure we didn't catch false positives. 

I was wary we might have a RNOH in the middle of a non RNOH reason for interaction (e.g. discharge not for patient going to RNOH). I'm happy either way.

Commit:
Show the send to EPR button if its not an RNOH reason for interaction

Prior to this change we were showing the send to EPR if the reason for
interaction did not start with RNOH, now we only show it if the reason
for interaction does not contain RNOH.

davidmiller commented:
Think we need to show the error by the save button - currently easy for it to be invalid but show no indication on screen...

<img width="620" alt="Screenshot 2023-01-11 at 13 30 15" src="https://user-images.githubusercontent.com/17229/211818688-e4e8f8b6-0fb9-459a-a23a-f5bbaddadaa7.png">

Commit:
Adds initials errors at the bottom of the micro input form

The microbiology input form is quite long, as such its easy to not see
why a form is not submitted. This puts the error message for initials,
max length or required, at the bottom
of the form.

Commit:
Adds initials errors at the bottom of the patient consultation form

The TB patient consultation form is quite long, so put the intials
errors at the bottom of the form so that they are always visible for the
user.

fredkingham commented:
good call, this is done and deployed to the test server.
<img width="901" alt="Screenshot 2023-01-12 at 11 46 35" src="https://user-images.githubusercontent.com/2175455/212059815-8c827446-3358-4234-b9bc-af6946c8e44a.png">

<img width="515" alt="Screenshot 2023-01-12 at 11 46 54" src="https://user-images.githubusercontent.com/2175455/212059821-0b23deea-1614-4c17-962d-1a7da37d1e0f.png">
